src/generator/backend/generator.js

```diff
@@ -0,0 +1,236 @@
+'use strict'
+
+var exports = module.exports = {};
+
+const fs = require('fs')
+const moment = require('moment')
+const handlebars = require('handlebars')
+
+const tpl = handlebars.compile(fs.readFileSync(__dirname + '/schedule.tpl').toString('utf-8'))
+const sessionsData = require('../../../mockjson/sessions.json')
+const speakersData = require('../../../mockjson/speakers.json')
+const servicesData = require('../../../mockjson/event.json')
+const sponsorsData = require('../../../mockjson/sponsors.json')
+
+if(!String.linkify) {
+  String.prototype.linkify = function() {
+    var urlPattern = /\b(?:https?|ftp):\/\/[a-z0-9-+&@#\/%?=~_|!:,.;]*[a-z0-9-+&@#\/%=~_|]/gim;
+    var pseudoUrlPattern = /(^|[^\/])(www\.[\S]+(\b|$))/gim;
+    var emailAddressPattern = /[\w.]+@[a-zA-Z_-]+?(?:\.[a-zA-Z]{2,6})+/gim;
+    return this
+            .replace(urlPattern, '<a href="$&">$&</a>')
+            .replace(pseudoUrlPattern, '$1<a href="http://$2">$2</a>')
+            .replace(emailAddressPattern, '<a href="mailto:$&">$&</a>');
+  };
+}
+
+handlebars.registerHelper('linkify', function (options) {
+  var content = options.fn(this);
+  return new handlebars.SafeString(content.linkify());
+});
+
+function slugify(str) {
+  if (str === undefined) {
+    return ""
+  }
+  return str.replace(/[^\w]/g, '-').replace(/-+/g, '-').toLowerCase()
+}
+
+function speakerNameWithOrg(speaker) {
+  return speaker.organisation ?
+    `${speaker.name} (${speaker.organisation})` :
+    speaker.name
+}
+
+function foldByDate(tracks) {
+  let dateMap = new Map()
+
+  tracks.forEach(track => {
+    if (!dateMap.has(track.date)) {
+      dateMap.set(track.date, {
+        caption: track.date,
+        tracks: []
+      })
+    }
+    dateMap.get(track.date).tracks.push(track)
+  })
+
+  let dates = Array.from(dateMap.values())
+  dates.forEach(date => date.tracks.sort(byProperty('sortKey')))
+
+  return dates;
+}
+
+function byProperty(key) {
+  return (a, b) => {
+    if (a[key] > b[key]) { return 1 }
+    if (a[key] < b[key]) { return -1 }
+    else { return 0 }
+  }
+}
+
+function zeroFill(num) {
+  if (num === undefined) {
+    return ""
+  }
+  if (num >= 10) { return num.toString() }
+  else { return '0' + num.toString() }
+}
+
+function foldByTrack(sessions, speakers) {
+  let trackData = new Map()
+  let speakersMap = new Map(speakers.map(s => [s.id, s]));
+
+  sessions.forEach(session => {
+    if (!session.start_time)
+      return
+
+    // generate slug/key for session
+    let date = moment(session.start_time).format('YYYY-MM-DD')
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'start_time' is not in camel case.](https://www.codacy.com/app/triplez/open-event-webapp/file/2705261309/issues/source?bid=3233857&fileBranchId=3342834#l89)